packages/flame/lib/src/components/position_component.dart

```diff
@@ -1,218 +1,326 @@
+    position += point - topLeftPosition;
+  }
 
-  /// Angle (with respect to the x-axis) this component should be rendered with.
-  /// It is rotated around its anchor.
-  double angle;
+  /// The position of the center of the component's bounding rectangle
+  /// in the parent's coordinates.
+  Vector2 get center => positionOf(Anchor.center);
+  set center(Vector2 point) {
+    position += point - center;
+  }
 
-  /// Anchor point for this component. This is where flame "grabs it".
-  /// The [position] is relative to this point inside the component.
-  /// The [angle] is rotated around this point.
-  Anchor anchor;
+  /// The [anchor]'s position in absolute (world) coordinates.
+  Vector2 get absolutePosition => absolutePositionOf(_anchor);
 
-  /// Whether this component should be flipped on the X axis before being rendered.
-  bool renderFlipX = false;
+  /// Get the absolute top left position regardless of whether it is a child or not
+  Vector2 get absoluteTopLeftPosition => absolutePositionOf(Anchor.topLeft);
 
-  /// Whether this component should be flipped ofn the Y axis before being rendered.
-  bool renderFlipY = false;
+  /// Get the absolute center of the component
+  Vector2 get absoluteCenter => absolutePositionOf(Anchor.center);
 
-  PositionComponent({
-    Vector2? position,
-    Vector2? size,
-    Vector2? scale,
-    this.angle = 0.0,
-    this.anchor = Anchor.topLeft,
-    this.renderFlipX = false,
-    this.renderFlipY = false,
-    int? priority,
-  })  : _position = position ?? Vector2.zero(),
-        _size = size ?? Vector2.zero(),
-        _scale = scale ?? Vector2.all(1.0),
-        super(priority: priority);
+  //----------------------------------------------------------------------------
+  // Mutators
+  //----------------------------------------------------------------------------
 
-  @override
-  bool containsPoint(Vector2 point) {
-    final rectangle = Rectangle.fromRect(toAbsoluteRect(), angle: angle)
-      ..position = absoluteCenter;
-    return rectangle.containsPoint(point);
+  /// Flip the component horizontally around its anchor point.
+  void flipHorizontally() => _transform.flipHorizontally();
+
+  /// Flip the component vertically around its anchor point.
+  void flipVertically() => _transform.flipVertically();
+
+  /// Flip the component horizontally around its center line.
+  void flipHorizontallyAroundCenter() {
+    final delta = (1 - 2 * _anchor.x) * width;
+    _transform.x += delta * math.cos(_transform.angle);
+    _transform.y += delta * math.sin(_transform.angle);
+    _transform.flipHorizontally();
+  }
+
+  /// Flip the component vertically around its center line.
+  void flipVerticallyAroundCenter() {
+    final delta = (1 - 2 * _anchor.y) * height;
+    _transform.x += -delta * math.sin(_transform.angle);
+    _transform.y += delta * math.cos(_transform.angle);
+    _transform.flipVertically();
   }
 
-  double angleTo(PositionComponent c) => position.angleTo(c.position);
```

Comment:
Why was this removed?

Reply:
Because this thing never actually worked. For example, if you have a component at (100, 100), and another component at (100, 110) then the angle from first to second is ~0.047 (or about 3 degrees). This has no geometrical meaning.

Now, I'm not against introducing some notion of an angle, but an angle requires two lines. The first line would be through the anchors of two components, but what about the other line? We can either pick some arbitrary fixed direction (like (0,1)), or make it a parameter to `angleTo`, or introduce a notion of "principal direction of a sprite" as an extra property of the PositionComponent.

But whatever we do, this needs a serious discussion, and perhaps can be considered as a new feature to be added. Right now, however, removing `angleTo` is justified because the current implementation doesn't work at all...

Reply:
Ah right, since the lines both now go from (0, 0).
What we should do is something like this then:
`(c.position - position).angleTo(Vector2(0, 1));`